### ISSUE TITLE
Upscaling doesnt work, gives me this error, how do i fix this? (input file is .mp4 and everything looks fine)

### ISSUE THREAD

**Issue by xytual** [Djdefrag/QualityScaler#104]
[QualityScaler] Preference file exist
Loading AI model
Upscaling from scrach
1. Extracting video audio
 Frames supported simultaneously by GPU: 47
1. Upscaling video
1. Upscaling video (6 threads)
1. Upscaling video 5.76% (06s)
1. Upscaling video 11.51% (04s)
1. Upscaling video 17.27% (03s)
1. Upscaling video 23.02% (02s)
1. Upscaling video 28.78% (02s)
1. Upscaling video 34.53% (02s)
1. Upscaling video 40.29% (01s)
1. Upscaling video 46.04% (01s)
1. Upscaling video 51.80% (01s)
1. Upscaling video 57.55% (01s)
1. Upscaling video 63.31% (01s)
1. Upscaling video 69.06% (01s)
1. Processing upscaled video
Error [Errno 22] Invalid argument

MoviePy error: FFMPEG encountered the following error while writing file C:/Users/Ben/Downloads/test_RealESR_Gx4_Resize-100_Interpolation-Low.mp4:

 b'[mp4 @ 000001deccbcdc00] Could not find tag for codec pcm_s16le in stream #1, codec not currently supported in container\r\nCould not write header for output file #0 (incorrect codec parameters ?): Invalid argument\r\nError initializing output stream 0:0 -- \r\n'

The video export failed, possibly because the codec specified for the video (libx264) is not compatible with the given extension (mp4). Please specify a valid 'codec' argument in write_videofile. This would be 'libx264' or 'mpeg4' for mp4, 'libtheora' for ogv, 'libvpx for webm. Another possible reason is that the audio codec was not compatible with the video codec. For instance the video extensions 'ogv' and 'webm' only allow 'libvorbis' (default) as avideo codec.

**Comment by Djdefrag** [comment]
Hi my friend, 

you need the newer version of ffmpeg.

1. Go to this site (https://www.gyan.dev/ffmpeg/builds/)
2. Download one of this zip
![365366956-6bbc7c1a-92ba-498c-9e02-fd858d2ba32b](https://github.com/user-attachments/assets/da527068-aeef-40d7-b149-a6a597c72fc0)
3. Extract ffmpeg.exe from the downloaded .zip
4. Copy ffmpeg.exe to QualityScaler/Assets

**Comment by eulisan** [comment]
Does not work for me, getting the same error with newest ffmpeg. But when i delete the audiofile from the "working" directory where also the frames and scaled images are stored, it works. 

Seems it cant process the audio into the video for me.

**Comment by Dodeedoo** [comment]
> Does not work for me, getting the same error with newest ffmpeg. But when i delete the audiofile from the "working" directory where also the frames and scaled images are stored, it works.
> 
> Seems it cant process the audio into the video for me.

you are correct, it is an issue with incorrect audio codec being used to save the video

add `audio_codec='pcm_32le'` replace 32 with bit depth of saved wav audio from your video

**Comment by eulisan** [comment]
Uhm, well. I used RealScaler which did extract the audio and works also well now. But i think it used external FFMPEG and i didnt had exiftool in the Asset Directory.

Now i placed FFMPEG and EXIFTOOL in the Asset Directory and using the  3.10 Qualityscaler it does work, but without audio. Means, i wont extract the audio.wav

But Realscaler 3.9 now works for me, lol.

**Comment by Djdefrag** [comment]
Exiftool only extract original file metadata (so it's not about audio)

With QualityScaler 3.10 the app does not extract the audio as file, but use FFMPEG to copy the audio from original video to upscaled video. This function is called Audio Passthrough
